R/utils.R

```diff
@@ -92,3 +92,123 @@ browse_url <- function(path, chromote) {
     utils::browseURL(url)
   }
 }
+
+
+# =============================================================================
+# Random Ports
+# =============================================================================
+#
+# Borrowed from https://github.com/rstudio/httpuv/blob/main/R/random_port.R
+
+with_random_port <- function(
+  startup,
+  min = 1024L,
+  max = 49151L,
+  n = 5,
+  host = "http://127.0.0.1"
```

Comment:
Should document that `host` must start with `http://` (or `https://`?) and not have a trailing slash. Or do a run-time check and throw an error with a useful message.

Reply:
Edit: now that I look at it some more, I don't think that the `host` parameter is necessary at all.

Reply:
I agree. That logic should be outside of `with_random_port()`, whose job should just be to run `startup()` with different `port` values.